src/shapes.cpp

```diff
@@ -298,40 +298,84 @@ void Sphere::scaleAndPadd(double scale, double padding)
   radius = tmpRadius;
 }
 
-void Cylinder::scaleAndPadd(double scale, double padding)
+void Cylinder::scaleAndPadd(double scaleRadius, double scaleLength, double paddRadius, double paddLength)
 {
-  const auto tmpRadius = radius * scale + padding;
-  const auto tmpLength = length * scale + 2.0 * padding;
+  const auto tmpRadius = radius * scaleRadius + paddRadius;
+  const auto tmpLength = length * scaleLength + 2.0 * paddLength;
   if (tmpRadius < 0 || tmpLength < 0)
     throw std::runtime_error("Cylinder dimensions must be non-negative.");
   radius = tmpRadius;
   length = tmpLength;
 }
 
-void Cone::scaleAndPadd(double scale, double padding)
+void Cylinder::scale(double scaleRadius, double scaleLength)
 {
-  const auto tmpRadius = radius * scale + padding;
-  const auto tmpLength = length * scale + 2.0 * padding;
+  scaleAndPadd(scaleRadius, scaleLength, 0.0, 0.0);
+}
+
+void Cylinder::padd(double paddRadius, double paddLength)
+{
+  scaleAndPadd(0.0, 0.0, paddRadius, paddLength);
```

Comment:
```suggestion
  scaleAndPadd(1.0, 1.0, paddRadius, paddLength);
```

I might be completely mistaken, but why did were these not caught by your tests? :confused: